src/main/java/org/monarchinitiative/fhir2hpo/loinc/DefaultLoinc2HpoAnnotation.java

```diff
@@ -104,44 +110,75 @@ public LoincId getLoincId() {
 	 * 
 	 * @return the LOINC scale
 	 */
+	@Override
 	public LoincScale getLoincScale() {
 		return loincScale;
 	}
 
 	@Override
-	public HpoTermWithNegation convert(Observation observation)
-			throws LoincException, ConversionException, FHIRException {
-
-		LoincId observationLoincId = ObservationUtil.getLoincIdOfObservation(observation);
-		if (!observationLoincId.equals(loincId)) {
-			throw new ConversionException(ConversionExceptionType.MISMATCHED_LOINC_ID, "Can only convert observations with LoincId " + loincId);
-		}
+	public HpoConversionResult convert(Observation observation) {
 		
-		if (observation.hasInterpretation()) {
-			Loinc2HpoCodedValue internalCode = CodeableConceptAnalyzer
-					.getInternalCodeForCodeableConcept(observation.getInterpretation());
-			return getHpoTermForInternalCode(internalCode);
+		HpoConversionResult result = new HpoConversionResult(observation);
+		try {
+			LoincId observationLoincId = ObservationUtil.getLoincIdOfObservation(observation);
+			if (!observationLoincId.equals(loincId)) {
+				throw new MismatchedLoincIdException("Can only convert observations with LoincId " + loincId);
+			}
+
+			result.addMethodConversionResult(convertInterpretation(observation));
+			result.addMethodConversionResult(convertValueQuantity(observation));
+			
+		} catch (Exception e) {
+			result.setException(e);
 		}
 
-		if (observation.hasValueQuantity()) {
-			return getHpoTermForValueQuantity(observation.getValueQuantity(), observation.getReferenceRange());
+		return result;
+	}
+
+	private MethodConversionResult convertInterpretation(Observation observation) {
+		MethodConversionResult result = new MethodConversionResult("Interpretation");
```

Comment:
It might be worth using an enum or class hierarchy rather than magic strings to set the `method` property.

Reply:
Sounds good.